ui/src/app/service/pipeline/pipeline.service.ts

```diff
@@ -23,8 +23,13 @@ export class PipelineService {
      * @param key Project unique key
      * @param pipName Pipeline Name
      */
-    getPipeline(key: string, pipName: string): Observable<Pipeline> {
-        return this._http.get('/project/' + key + '/pipeline/' + pipName).map(res => res.json());
+    getPipeline(key: string, pipName: string, withApplications: boolean = false): Observable<Pipeline> {
```

Comment:
i think it will be easier to manage cache if getPipeline return Always thé same content, like getProject and getApplication. If not each time WE will have toi check if content we need is in cache.